modules/core/src/types/layer-props.ts

```diff
@@ -75,6 +75,19 @@ export type Texture =
   | HTMLVideoElement
   | ImageBitmap;
 
+/**
+ * Default type of `data` property. Can be either
+ * `LayerData` - the required data type is supplied
+ * `string` it's interpreted as URL which is fetched and parsed (if a matching loader is found). The loader is  expected to return the compatible layer data type.
+ * A `Promise` is expected to resolve to `LayerData` type.
+ * An `AsyncIterable` is expected to deliver chunks of data.
+ */
+export type LayerDataProp<DataType = any> =
+  | LayerData<DataType>
+  | string
+  | AsyncIterable<DataType>
```

Comment:
This is actually a bug - it should be `AsyncIterable<DataType[]>` (resolves to a chunk, not individual datum)